paperqa/agents/search.py

```diff
@@ -443,9 +445,13 @@ async def get_directory_index(
 
     valid_files = [
         file
-        async for file in directory.iterdir()
+        async for file in (directory.rglob("*") if recursive else directory.iterdir())
         if file.suffix in {".txt", ".pdf", ".html"}
     ]
+    if len(valid_files) > 999:  # noqa: PLR2004
```

Comment:
Why `999` and can you move it to a module level or class variable?

```python
VALID_FILE_LENGTH = 999
```

Reply:
Made a module-level variable. And it's an arbitrary cutoff